### PR TITLE
dlgDotplot

### PR DESCRIPTION
**Pull request by maxwellfundi**
@africanmathsinitiative/developers This is ready for review

### CODE REVIEW

File: instat/dlgDotPlot.vb [hunk 1]
```diff
@@ -29,134 +31,194 @@ Public Class dlgDotPlot
...
+        ucrOtherAxisReceiver.SetParameter(New RParameter("y"))
+        ucrOtherAxisReceiver.bWithQuotes = False
+        ucrOtherAxisReceiver.SetValuesToIgnore({Chr(34) & Chr(34)})
+        ucrOtherAxisReceiver.bAddParameterIfEmpty = True
+        ucrOtherAxisReceiver.SetParameterIsString()
+
         ucrFactorReceiver.Selector = ucrDotPlotSelector
         ucrFactorReceiver.SetIncludedDataTypes({"factor"})
-        ucrBase.clsRsyntax.iCallType = 0
+        ucrFactorReceiver.SetParameter(New RParameter("fill"))
```
Comment (lilyclements) [inline instat/dlgDotPlot.vb]: We need parameter positions

File: instat/dlgDotPlot.vb [hunk 2]
```diff
@@ -29,134 +31,194 @@ Public Class dlgDotPlot
...
-            SetBinAxisAes()
-        End If
+        ''This sub changes the choice of bin axis.
+        'If rdoYBinAxis.Checked Then
+        '    strBinAxis = "y"
+        '    strOtherAxis = "x"
+        'Else
+        '    strBinAxis = "x"
+        '    strOtherAxis = "y"
+        'End If
```
Comment (lilyclements) [inline instat/dlgDotPlot.vb]: How come all of this is commented out? Is it temporarily not needed or can it be removed?
Perhaps if temporary, you could put a comment explaining?

File: instat/dlgDotPlot.vb [hunk 3]
```diff
@@ -213,7 +266,15 @@ Public Class dlgDotPlot
...
         TestOkEnabled()
     End Sub
+
+    Private Sub ucrVariablesAsFactorDotPlot_SelectionChanged(sender As Object, e As EventArgs) Handles ucrVariablesAsFactorDotPlot.SelectionChanged
+
+    End Sub
+
+    Private Sub ucrDotPlotSelector_DataFrameChanged()
+
+    End Sub
```
Comment (lilyclements) [inline instat/dlgDotPlot.vb]: Empty sub can be deleted

Comment (maxwellfundi) [inline instat/dlgDotPlot.vb]: This will be fixed thanks.

**Comment by maxwellfundi** [comment]
The x and y binaxis radio buttons changed because, i figured out it would look better and consistent with our new rdo as buttons 
If this is not appropriate i could get back to the radio buttons

File: instat/dlgDotPlot.vb (hunk 1 above)
Comment (maxwellfundi) [inline instat/dlgDotPlot.vb]: Okay... I will get that sorted too.

File: instat/dlgDotPlot.vb (hunk 2 above)
Comment (maxwellfundi) [inline instat/dlgDotPlot.vb]: We dont need this for now. with the new linking system, the controls know when and how to add what

Comment (lilyclements) [inline instat/dlgDotPlot.vb]: I see. Why can this code not be removed?

Comment (maxwellfundi) [inline instat/dlgDotPlot.vb]: it can actually be removed but I chose to leave it for now

**Comment by lilyclements** [comment]
I see, I think in this dialog it works better to have the binaxis as radio buttons not button buttons. This is because I think button-buttons work when the base R-code is changing which isn't in this instance. This is a minor change to a parameter

**Comment by dannyparsons** [comment]
We should use button rdos when it is a main option which changes the whole dialog, and is therefore usually at the top. For example, a different plot or different function which changes how you might use the remaining controls.

For normal options on a dialog we should keep the circular rdos

**Comment by lilyclements** [comment]
I'm planning to edit the design of this dialog so I can fix that

File: instat/dlgDotPlot.vb [hunk 4]
```diff
@@ -29,134 +31,194 @@ Public Class dlgDotPlot
...
         ucrDotPlotSelector.Reset()
-        ucrSaveDotPlot.strPrefix = "Dotplot"
         ucrSaveDotPlot.Reset()
         sdgPlots.Reset()
-        rdoXBinAxis.Checked = True 'If it was already True, no need to change anything, else it will raise event rdoBinAxis_CheckChanged
-        SetFactorAxisAes()
-        SetBinAxisAes() 'TestOk is done within this sub.
+
+        'I am not sure we need this
+        bEditAesFunction = True
```
Comment (lilyclements) [inline instat/dlgDotPlot.vb]: You've put you're unsure if we need this so perhaps we should address this. What does this boolean do?

**Comment by maxwellfundi** [comment]
@lilyclements  i have removed the unnecessary code now. This is now ready for @dannyparsons

File: instat/dlgDotPlot.vb [hunk 5]
```diff
@@ -17,10 +17,12 @@
 Imports instat.Translations
 Public Class dlgDotPlot
     'In order, the global ggplot function, the geom_dotplot function and the aes function, used as parameter in the ggplot function.
-    Private clsRggplotFunction As New RFunction
-    Private clsRgeom_dotplot As New RFunction
+    Private clsRggPlotFunction As New RFunction
+    Private clsRgeom_DotPlot As New RFunction
```
Comment (dannyparsons) [inline instat/dlgDotPlot.vb]: I don't like these names very much. Can we have something that's more consistent with other variables?

File: instat/dlgDotPlot.vb [hunk 6]
```diff
@@ -29,145 +31,127 @@ Public Class dlgDotPlot
...
-            strOtherAxis = "y"
-        End If
-        clsRgeom_dotplot.AddParameter("binaxis", Chr(34) & strBinAxis & Chr(34))
-        'The aes parameters for x and y in the AesFunction need to be updated, unless we are setting up the dialogue according to the content of AesFunction, when coming back from LayerOptions.
-        If bEditAesFunction Then
-            SetFactorAxisAes()
-            SetBinAxisAes()
-        End If
-    End Sub
+        'I am not sure we need this
```
Comment (dannyparsons) [inline instat/dlgDotPlot.vb]: Which part?

Comment (maxwellfundi) [inline instat/dlgDotPlot.vb]: This is okay. I know where its used and we can keep it for now.

**Pull request merged**